modules/configuration-api/src/main/java/org/apache/ignite/configuration/annotation/ConfigurationExtension.java

```diff
@@ -39,5 +40,12 @@
 @Target(TYPE)
 @Retention(RUNTIME)
 @Documented
-public @interface InternalConfiguration {
+public @interface ConfigurationExtension {
```

Comment:
I don't like replacing one annotation with another, it's better to add a new one.